drivers/hwinfo/Kconfig

```diff
@@ -32,6 +32,7 @@ config HWINFO_NRF
 	bool "NRF device ID"
 	default y
 	depends on SOC_FAMILY_NRF
+	depends on SOC_SECURE_NRF
```

Comment:
Because this is a working solution for the problem that I want to solve, which is to support other implementations of this interface other than TF-M.